core/domain/stats_jobs_continuous.py

```diff
@@ -617,20 +618,23 @@ def reduce(key, stringified_values):
         # the map function). For this reason, all of this code is simply skipped
         # in that case in favor of performance.
         if len(versions) > 1:
-            start_version_index = next(
-                (index for index, version in enumerate(versions[1:])
-                 if (versioned_interaction_ids[version][0]
-                     != latest_interaction_id)),
-                len(versions) - 1)
-            start_version = versions[start_version_index]
+            indexes = [
+                index for index, version in enumerate(versions[1:])
+                if versioned_interaction_ids[version][0] != (
+                    latest_interaction_id)]
+            earliest_acceptable_version_index = (
+                indexes[0] if indexes else len(versions) - 1)
```

Comment:
OK -- the -1 is indeed terrible.

How about:

    invalid_version_indexes = [index for index, version in enumerate(versions) if ...]
    earliest_acceptable_version_index = (invalid_version_indexes[0] - 1 if indexes else len(versions) - 1)